Upsteam-Status Backport [fixed in 7.37.0]
commit 38d582ff541353d738858299d4a2b78bafac03ed
Author: Daniel Stenberg <daniel@haxx.se>
Date:   Wed Mar 26 13:22:10 2014 +0100

    mkhelp: generate code for --disable-manual as well
    
    This allows configure --disable-manual to run and build without having
    to regenerate the src/tool_hugehelp.c file which otherwise is necessary
    since we ship tarballs with that file present.
    
    Reported-by: Remi Gacogne
    Bug: http://curl.haxx.se/bug/view.cgi?id=1350

diff --git a/src/mkhelp.pl b/src/mkhelp.pl
index 444b669..7ed86f7 100644
--- a/src/mkhelp.pl
+++ b/src/mkhelp.pl
@@ -6,7 +6,7 @@
 #                            | (__| |_| |  _ <| |___
 #                             \___|\___/|_| \_\_____|
 #
-# Copyright (C) 1998 - 2011, Daniel Stenberg, <daniel@haxx.se>, et al.
+# Copyright (C) 1998 - 2014, Daniel Stenberg, <daniel@haxx.se>, et al.
 #
 # This software is licensed as described in the file COPYING, which
 # you should have received as part of this distribution. The terms
@@ -256,6 +256,10 @@ foot();
 
 sub foot {
   print <<FOOT
+#else /* !USE_MANUAL */
+/* built-in manual is disabled, blank function */
+#include "tool_hugehelp.h"
+void hugehelp(void) {}
 #endif /* USE_MANUAL */
 FOOT
   ;
